envoy/grpc/async_client_manager.h

```diff
@@ -15,11 +17,21 @@ class AsyncClientFactory {
 
   /**
    * Create a gRPC::RawAsyncClient.
+   * Prefer AsyncClientManager::getOrCreateRawAsyncClient() to creating uncached raw async client
+   * from factory directly. Only call this method when the raw async client must be owned
+   * exclusively. For example, some filters pass *this reference to raw client. In this case, the
+   * client must be destroyed before the filter instance. In this case, the grpc client must be
+   * owned by the filter instance exclusively.
    * @return RawAsyncClientPtr async client.
    */
-  virtual RawAsyncClientPtr create() PURE;
+  virtual RawAsyncClientPtr createUncachedRawAsyncClient() PURE;
+
+private:
+  friend class AsyncClientFactoryImpl;
 };
 
+enum class CacheOption { AlwaysCache, CacheWhenRuntimeEnabled };
```

Comment:
Is this only for test? If so, can we just use runtime overrides in tests?

Reply:
The runtime and the cache option decide the cache behavior together, with always cache option the async client will always be cached even when runtime is disabled. If the option is cached when runtime enabled, will only cache when runtime is enable.

Reply:
so when the runtime is disabled, only ex_authz will use the async client cache, which will degrade to the old behavior.

Reply:
OK, so this will disappear when runtime is eventually removed? Maybe leave a comment.

Reply:
yes, will leave a TODO here.

Reply:
Nit: s/TO/TODO/